pkg/auth/auth_handler.go

```diff
@@ -179,26 +179,34 @@ func (a *AuthHandler) AuthUserAuth(token string) (interface{}, error) {
 		return nil, err
 	}
 
+	if payload.Username == "" {
+		a.log.Error("Missing username in token")
+		return nil, fmt.Errorf("Missing username in token")
+	}
+
 	err = a.storeAdminInPayload(payload)
 	if err != nil {
 		a.log.Errorf("Unable to fetch user's capabilities: %v", err)
 		return nil, common.ApiErrorWithDefaultInfraError(err, http.StatusUnauthorized)
 	}
 
-	if payload.Username == "" {
-		a.log.Error("Missing username in token")
-		return nil, fmt.Errorf("Missing username in token")
-	}
-
 	return payload, nil
 }
 
 func (a *AuthHandler) storeAdminInPayload(payload *ocm.AuthPayload) error {
+	payloadKey := payload.Username + "_is_admin"
```

Comment:
Is `_is_admin` used somewhere?

Reply:
Just here, it's only a postfix to make the key unique.

Reply:
Then when you read it don't you need to get this suffix as well?